x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/StopWatch.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.profiling;
+
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Measures time and logs it in milliseconds.
+ */
+public class StopWatch {
+    private final String name;
+    private final long start;
+
+    public StopWatch(String name) {
+        this.name = name;
+        start = System.nanoTime();
+    }
+
+    /**
+     * Log name and the number of elapsed milliseconds since object creation.
+     *
+     * @param log Logger for logging the report.
+     */
+    public void Log(Logger log) {
```

Comment:
Can we remove this method and rely entirely on `Report`? This would simplify the class.